### ISSUE TITLE
Minimise duplicate code between roles and tasks

### ISSUE THREAD

**Issue by SamLubbers** [xrnl/aginun#34]
There is a lot of duplicate code for similar functionalities of roles and tasks. In particular:
* `RolesOverview.vue` and `TaskOverview.vue`
* `RoleFilter.vue` and `TaskFilter.vue`
* `NewRoleDialog.vue` and `NewTaskDialog.vue`
* `RoleViewDialog.vue` and `TaskViewDialog.vue`
To make the code more maintainable it would be useful to replace this duplicate code with code that is reused between roles and tasks.

**Comment by moritzcopter** [comment]
I thought about this one a bit - wouldnt we make our lives easier if we would only have one task/role type (frontend but also in the db) and instead just have a field "IsTask" that the user can filter on?

This also reduces complexity for the user as they don't have to understand that there are two dashboards and then click through both of them to find something they are interested in.

**Comment by SamLubbers** [comment]
Hey @moritzcopter, thank you for giving this some thought 💚  I agree with you that having one dashboard makes it easier to find ways to volunteer. For example, if someone wants to volunteer with `XR Amsterdam` they should only filter once for that local group and see all available roles and tasks (i.e. all available volunteering opportunities). We should also add a filter if people are looking for only roles or only tasks.

I think that in the backend it is beneficial to have separate tables for [roles](https://github.com/xrnl/aginun/wiki/Roles:-why,-what-and-how#role-details) and [tasks](https://github.com/xrnl/aginun/wiki/Tasks:-why,-what-and-how#task-details) because each of them (will) have unique fields. For example, roles require the `responsibilities` field but tasks do not. If we want to add tasks to the same table as roles, we'll have to make `responsibilities` an optional field, which isn't great because it won't reject roles that are submitted without any responsibilities. Also, if we use one table for both, we'll have a lot of empty values for fields that are unique to either roles or tasks. 

In conclusion, for refactoring the code, I think we can
  * Have one overview page with loads both roles and tasks
  * Have one filter for both roles and tasks
  * Share most code for the view and edit dialog, and like you said, use an `IsTask` variable to conditionally render parts of the component that are unique to either tasks or roles. 

What do you think? 🤔
